distribution/lib/Standard/Examples/0.0.0-dev/src/Main.enso

```diff
@@ -18,11 +18,11 @@ type Example_Error_Type
 
 ## The standard library data directory.
 data_dir : File
-data_dir = enso_project.data
+data_dir = (enso_project Standard.Examples).data
```

Comment:
Why is this change needed?

I assume the default should work OK here.

Reply:
Hmmm. So this is a change of semantics from what was there before?

I'm slightly worried about this change. It does align with how the new syntax work, but from user perspective - now it's basically never a good idea to call `enso_project` without specifying the argument explicitly - because as you've shown - _now_ if I don't specify the argument, the result of such a method will depend on _where it is called from_. I'm really not sure if this method's result should depend on the runtime call-site and not where 'statically' it is in the code.

I expect at some point when people start using `enso_project` in places other than directly in the GUI, it may very easily bite us.

Maybe we should keep `enso_project` without arguments and note that it returns the _currently opened project regardless of where it is called in_? (i.e. in the IDE it would always return the current project, even if called in some other module - i.e. it would work like the current default behaviour)

And then we could have a separate helper `find_project_for_module` taking the argument?

I feel like having both these functionalities in a single method may lead to mistakes.

Reply:
> `enso_project` without any arguments will resolve to the current project from the "statically" perspective

Get the call stack. Walk-up one frame. Find caller. Find its `RootNode`. If it is `EnsoRootNode` call its `getModuleScope()`. Cache the value (per `EnsoContext`) and make it compile time constant (at least for a finite number of contexts).

> this is a change of semantics from what was there before?

It is preferable to avoid avoid changes of semantics.

Reply:
> Get the call stack. Walk-up one frame. Find caller. Find its RootNode. If it is EnsoRootNode call its getModuleScope().

Done in 005fb3807bd8590b35baea2c7e7e72f48f1c6f3e

> Cache the value (per EnsoContext) and make it compile time constant (at least for a finite number of contexts).

What do you mean exactly by caching the value per `EnsoContext`? The Project Description depends on the location of the caller in the source code, not on the context. In one `EnsoContext`, there might be multiple callers from different modules. Am I misunderstanding your proposition?